contracts/PriceFeed/PriceFeed.sol

```diff
@@ -36,37 +51,59 @@ contract PriceFeed is IPriceFeed, Initializable, ProxyOwned {
         uint8 decimals = aggregator.decimals();
         require(decimals <= 18, "Aggregator decimals should be lower or equal to 18");
         if (address(aggregators[currencyKey]) == address(0)) {
-            aggregatorKeys.push(currencyKey);
+            currencyKeys.push(currencyKey);
         }
         aggregators[currencyKey] = aggregator;
         currencyKeyDecimals[currencyKey] = decimals;
         emit AggregatorAdded(currencyKey, address(aggregator));
     }
 
+    function addPool(bytes32 currencyKey, address poolAddress) external onlyOwner {
+        IUniswapV3Pool pool = IUniswapV3Pool(poolAddress);
```

Comment:
we should have stricter rules on adding a pool if the currencyKey already has an aggregator.